packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -253,54 +344,58 @@ export const useArbTokenBridge = (
     updateTokenData(erc20L1Address)
   }
 
-  const depositToken = useCallback(
-    async (erc20L1Address: string, amountRaw: BigNumber) => {
-      const bridgeToken = bridgeTokens[erc20L1Address]
-      const { symbol, decimals } = await (async () => {
-        if (bridgeToken) {
-          const { symbol, decimals } = bridgeToken
-          return { symbol, decimals }
-        }
-        const { symbol, decimals } = await bridge.l1Bridge.getL1TokenData(
-          erc20L1Address
-        )
-        addToken(erc20L1Address)
-        return { symbol, decimals }
-      })()
-      const amountReadable = await utils.formatUnits(amountRaw, decimals)
+  async function depositToken(erc20L1Address: string, amount: BigNumber) {
+    if (typeof l2Signer === 'undefined') {
+      throw new Error(`No instance of L2Signer found`)
+    }
 
-      const tx = await bridge.deposit({
-        erc20L1Address: erc20L1Address,
-        amount: amountRaw
-      })
+    if (typeof l2Signer.provider === 'undefined') {
+      throw new Error(`No instance of L2Signer found`)
+    }
 
-      addTransaction({
-        type: 'deposit-l1',
-        status: 'pending',
-        value: amountReadable,
-        txID: tx.hash,
-        assetName: symbol,
-        assetType: AssetType.ERC20,
-        sender: await walletAddressCached(),
-        l1NetworkID: await l1NetworkIDCached()
-      })
-      try {
-        const receipt = await tx.wait()
-        const seqNums = await bridge.getInboxSeqNumFromContractTransaction(
-          receipt
-        )
-        if (!seqNums) throw new Error('No sequence number detected')
-        const seqNum = seqNums[0].toNumber()
-        updateTransaction(receipt, tx, seqNum)
-        updateTokenData(erc20L1Address)
-        return receipt
-      } catch (err) {
-        console.warn('deposit token failure', err)
-        throw err
+    if (typeof erc20Bridger === 'undefined') {
+      throw new Error(`No instance of Erc20Bridger found`)
+    }
+
+    const { symbol, decimals } = await getL1TokenData(erc20L1Address)
+
+    const tx = await erc20Bridger.deposit({
+      l1Signer,
+      l2Provider: l2Signer.provider,
+      erc20L1Address,
+      amount
+    })
+
+    addTransaction({
+      type: 'deposit-l1',
+      status: 'pending',
+      value: utils.formatUnits(amount, decimals),
+      txID: tx.hash,
+      assetName: symbol!,
+      assetType: AssetType.ERC20,
+      // TODO: Update the following two calls
+      sender: await walletAddressCached(),
+      l1NetworkID: await l1NetworkIDCached()
+    })
+
+    try {
+      const receipt = await tx.wait()
+      const messages = await receipt.getL1ToL2Messages(l2Signer)
+
+      if (messages.length === 0) {
```

Comment:
we might as well insist on exactly 1 message here (and throw a scary "contact support" error if not)